### ISSUE TITLE
ty not allowing passing `Any` as function default.

### ISSUE THREAD

**Issue by Apakottur** [astral-sh/ty#3311]
### Summary

Consider the following code:
```python
# a.py

from typing import Any


class _Unset(Any):
    pass


UNSET: _Unset = _Unset()


def func(my_var: bool | None = UNSET) -> None:
    print(my_var)
```

Both mypy and pyright allow it, but ty does not:
```text
$ mypy a.py 
Success: no issues found in 1 source file
$ pyright a.py 
0 errors, 0 warnings, 0 informations
$ ty check a.py 
error[invalid-parameter-default]: Default value of type `_Unset` is not assignable to annotated parameter type `bool | None`
  --> a.py:13:10
   |
13 | def func(my_var: bool | None = UNSET) -> None:
   |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |

Found 1 diagnostic
```

We use this pattern to distinguish between a `None` value and an argument not being passed to the function at all.
Is this a bug in `ty` or mypy+pyright are both wrong?

### Version

0.0.32

**Comment by carljm** [comment]
ty's behavior here is intentional, but it's hard to say who is wrong or right. Mypy/pyright interpret "inherit from Any" as "acts like Any"; ty interprets it as "has an Any base class". This difference shows up when assigning it to a final type like `bool`. Since `bool` cannot have subclasses, it is impossible for `_Unset` to be a subtype of `bool`, no matter what the `Any` might materialize to.

This is tracked in #1686

**Comment by Apakottur** [comment]
Thanks @carljm , this makes sense.
Is there a differnet pattern I could use for this behavior (distinguish None and Unset) that ty allows?

**Comment by carljm** [comment]
@Apakottur I guess it depends what your criteria are for a good solution. For example, your current approach is unsafe/unsound, in the sense that the typechecker won't help you ensure that you handle the `UNSET` possibility inside the body of the function; you are effectively lying to the type checker and claiming that the only possible values are `bool | None`. For me, I would prefer a solution that isn't unsound on this way, and that actually doesn't require any fancy solutions or inheriting `Any`; you'd just annotate the parameter as `bool | None | _Unset`.
